Simbody/tests/TestParallelForces.cpp

```diff
@@ -0,0 +1,110 @@
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include "SimTKsimbody.h"
+#include <time.h>
+#include <chrono>
+#include <thread>
+#include <iostream>
+
+using namespace SimTK;
+using namespace std;
+
+class ParallelForceImpl : public Force::Custom::Implementation {
+public:
+    ParallelForceImpl() {}
+    bool shouldBeParallelIfPossible() const override{
+      return true;
+    }
+    void calcForce(const State& state, Vector_<SpatialVec>& bodyForces,
+          Vector_<Vec3>& particleForces, Vector& mobilityForces) const override{
+         std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+    Real calcPotentialEnergy(const State& state) const override{
+        return 0.0;
+    }
+};
+
+class NonParallelForceImpl : public Force::Custom::Implementation {
+public:
+    NonParallelForceImpl() {}
+    bool shouldBeParallelIfPossible() const override{
+      return false;
+    }
+    void calcForce(const State& state, Vector_<SpatialVec>& bodyForces,
+          Vector_<Vec3>& particleForces, Vector& mobilityForces) const override{
+         std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+    Real calcPotentialEnergy(const State& state) const override{
+        return 0.0;
+    }
+};
+
+void testParallelForce()
+{
+    MultibodySystem system;
+    SimbodyMatterSubsystem matter(system);
+    GeneralForceSubsystem forces(system);
+    
+    for(int x = 0; x < 50; x++)
+      Force::Custom custom(forces, new ParallelForceImpl());
+    
+    system.realizeTopology();
+    State state = system.getDefaultState();
+    system.realize(state, Stage::Dynamics);
+}
+
+void testNonParallelForce()
+{
+    MultibodySystem system;
+    SimbodyMatterSubsystem matter(system);
+    GeneralForceSubsystem forces(system);
+    
+    for(int x = 0; x < 50; x++)
+      Force::Custom custom(forces, new NonParallelForceImpl());
+    
+    system.realizeTopology();
+    State state = system.getDefaultState();
+    system.realize(state, Stage::Dynamics);
+}
+
+int main()
+{
+    SimTK_START_TEST("TestParallelForces");
+        auto start = std::chrono::high_resolution_clock::now();
+        testParallelForce();
+        auto end = std::chrono::high_resolution_clock::now();
+        std::chrono::duration<double, std::milli> elapsedParallel = end-start;
+        
+        start = std::chrono::high_resolution_clock::now();
+        testNonParallelForce();
+        end = std::chrono::high_resolution_clock::now();
+        std::chrono::duration<double, std::milli> elapsedNonParallel = end-start;
+        
+        cout << "Parallel Forces Time: " << elapsedParallel.count() << endl;
+        cout << "NonParallel Forces Time: " << elapsedNonParallel.count() << endl;
+        
+        SimTK_TEST(elapsedNonParallel.count() > elapsedParallel.count());
```

Comment:
Yep, I added in an if statement (line 95) to just pass the test if only one thread will be launched. (See the latest commit)